op-node/p2p/cli/load_config.go

```diff
@@ -0,0 +1,301 @@
+		}
+		if err != nil {
+			return fmt.Errorf("failed to make %s constructor: %w", v, err)
+		}
+		conf.HostSecurity = append(conf.HostSecurity, sc)
+	}
+
+	conf.PeersLo = ctx.GlobalUint(flags.PeersLo.Name)
+	conf.PeersHi = ctx.GlobalUint(flags.PeersHi.Name)
+	conf.PeersGrace = ctx.GlobalDuration(flags.PeersGrace.Name)
+	conf.NAT = ctx.GlobalBool(flags.NAT.Name)
+	conf.UserAgent = ctx.GlobalString(flags.UserAgent.Name)
+	conf.TimeoutNegotiation = ctx.GlobalDuration(flags.TimeoutNegotiation.Name)
+	conf.TimeoutAccept = ctx.GlobalDuration(flags.TimeoutAccept.Name)
+	conf.TimeoutDial = ctx.GlobalDuration(flags.TimeoutDial.Name)
+
+	peerstorePath := ctx.GlobalString(flags.PeerstorePath.Name)
+	if peerstorePath == "" {
+		return errors.New("peerstore path must be specified, use 'memory' to explicitly not persist peer records")
+	}
+
+	var err error
+	var store ds.Batching
+	if peerstorePath == "memory" {
+		store = sync.MutexWrap(ds.NewMapDatastore())
+	} else {
+		store, err = leveldb.NewDatastore(peerstorePath, nil) // default leveldb options are fine
+		if err != nil {
+			return fmt.Errorf("failed to open leveldb db for peerstore: %w", err)
+		}
+	}
+	conf.Store = store
+
+	return nil
+}
+
+func loadNetworkPrivKey(ctx *cli.Context) (*crypto.Secp256k1PrivateKey, error) {
+	raw := ctx.GlobalString(flags.P2PPrivRaw.Name)
+	if raw != "" {
+		return parsePriv(raw)
+	}
+	keyPath := ctx.GlobalString(flags.P2PPrivPath.Name)
+	f, err := os.OpenFile(keyPath, os.O_RDONLY, 0600)
+	if os.IsNotExist(err) {
+		p, _, err := crypto.GenerateSecp256k1Key(rand.Reader)
+		if err != nil {
+			return nil, fmt.Errorf("failed to generate new p2p priv key: %w", err)
+		}
+		b, err := p.Raw()
+		if err != nil {
+			return nil, fmt.Errorf("failed to encode new p2p priv key: %w", err)
+		}
+		f, err := os.OpenFile(keyPath, os.O_CREATE|os.O_WRONLY, 0600)
+		if err != nil {
+			return nil, fmt.Errorf("failed to store new p2p priv key: %w", err)
+		}
+		defer f.Close()
+		if _, err := f.WriteString(hex.EncodeToString(b)); err != nil {
+			return nil, fmt.Errorf("failed to write new p2p priv key: %w", err)
+		}
+		return (p).(*crypto.Secp256k1PrivateKey), nil
+	} else {
+		defer f.Close()
+		data, err := io.ReadAll(f)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read priv key file: %w", err)
+		}
+		return parsePriv(strings.TrimSpace(string(data)))
+	}
+}
+
+func parsePriv(data string) (*crypto.Secp256k1PrivateKey, error) {
+	if len(data) > 2 && data[:2] == "0x" {
+		data = data[2:]
+	}
+	b, err := hex.DecodeString(data)
+	if err != nil {
+		return nil, errors.New("p2p priv key is not formatted in hex chars")
+	}
+	p, err := crypto.UnmarshalSecp256k1PrivateKey(b)
+	if err != nil {
+		// avoid logging the priv key in the error, but hint at likely input length problem
+		return nil, fmt.Errorf("failed to parse priv key from %d bytes", len(b))
+	}
+	return (p).(*crypto.Secp256k1PrivateKey), nil
+}
+
+func loadGossipOptions(conf *p2p.Config, ctx *cli.Context) error {
```

Comment:
Any sanity checks in here for values that make no sense? Like setting the lo flag higher than the hi flag? Or setting them to 0? Since the no discovery flag should turn it off, that should be used when wanting to configure it off and the values passed here should always be able to work

Reply:
These are checked in `Config.Check()`

Reply:
Ok i see the sanity checks happen in a `Check()` function